File: sorting-algorithms/insertion-sort/C/insertion-sort.c
#include <stdio.h>

void insertionSort(int[], int);

int main()
{

    int arr[4] = {8, 6, 3, 2};

    insertionSort(arr, 4);

    for (int i = 0; i < 4; ++i)
    {
        printf("%d ", arr[i]);
    }

    return 0;
}

void insertionSort(int arr[], int size)
{
    for (int i = 1; i < size; ++i)
    {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key)
        {
            arr[j + 1] = arr[j];
            --j;
        }
        arr[j + 1] = key;
    }
}

File: algorithms/sorting-algorithms/insertion-sort/JS/insertion-sort.js
function insertionSort(arr){
    const size = arr.length;
    const sorted = [...arr];

    for(let i = 1; i < size; ++i){
        let j = i - 1;
        let key = sorted[i];

        while(j >= 0 && sorted[j] > key){
            sorted[j + 1] = sorted[j];
            --j;
        }

        sorted[j + 1] = key;
    }
    return sorted;
}

let arr = [8,3,5,2];

let sorted = insertionSort(arr);
console.log(sorted);